### PR TITLE
MGMT-10946: fix memory usage based on OCP requirements

### PR DESCRIPTION
**Pull request by osherdp**
* Update all SNO installations (defined in assisted-test-infra at least) to use exactly 16GiB of RAM as official requirement dictates. It should apply to e2e-metal-assisted-single-node job and for kube-api jobs.
* Update other memory definitions of master and worker nodes to be actual multiples of gigibytes.

### CODE REVIEW

File: src/consts/resources.py [hunk 1]
```diff
@@ -1,12 +1,12 @@
 from consts import DISK_SIZE_100GB, GB
 
-DEFAULT_WORKER_MEMORY: int = 8892
-DEFAULT_MASTER_MEMORY: int = 16984
+DEFAULT_WORKER_MEMORY: int = 8 * 1024  # mebibytes
```
Comment (eliorerz) [inline src/consts/resources.py:3]: Why not declare `MiB = 1024` as same as we did on `GB`?

Comment (eliorerz) [reply src/consts/resources.py:3]: That will make all the comments redundant

**Review by eliorerz — CHANGES_REQUESTED** [review]

File: src/consts/resources.py (hunk 1 above)
Comment (osherdp) [reply src/consts/resources.py:3]: ``MiB = 1024`` seems incorrect, because 1024 is actually K

Comment (osherdp) [reply src/consts/resources.py:3]: I initially did ``(8 * GiB) // MiB`` or something alike, not sure what's more clear of the two options

Comment (eliorerz) [reply src/consts/resources.py:3]: Maybe something like `DEFAULT_WORKER_MEMORY: int = 3 * MiB_UNITS` where `MiB_UNITS=1024`?

Comment (osherdp) [reply src/consts/resources.py]: Done!

**Comment by osherdp** [comment]
AFAIK conformance tests failure in [e2e-metal-single-node-with-worker-live-iso](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_assisted-test-infra/1732/pull-ci-openshift-assisted-test-infra-master-e2e-metal-single-node-with-worker-live-iso/1545038081113985024) are known issue

**Review by eliorerz — APPROVED** [review]
LGTM

**Comment by osherdp** [comment]
/test e2e-metal-assisted-odf
seems like the existing flake we know about. retesting

**Comment by osherdp** [comment]
/override ci/prow/e2e-metal-single-node-with-worker-live-iso
known issue

**Pull request merged**